### PR TITLE
Clean up logic in the fatal error emitting code

### PR DESCRIPTION
**Pull request by emaxx-google**
Make the code more straightforward and also handle the (potential) case
when the goog.asserts.ENABLE_ASSERTS constant is changed from its default
value.

### CODE REVIEW

**Comment by emaxx-google** [comment]
Ivan, PTAL.

File: common/js/src/logging/logging.js [hunk 1]
```diff
@@ -195,14 +194,13 @@ GSC.Logging.checkWithLogger = function(
  * @param {...*} var_args The items to substitute into the failure message.
  */
 GSC.Logging.fail = function(opt_message, var_args) {
-  var messageAndArgs = Array.prototype.slice.call(arguments);
   if (goog.DEBUG) {
-    goog.asserts.fail.apply(goog.asserts, messageAndArgs);
+    throwAssertionError(opt_message, var_args);
   } else {
+    var messageAndArgs = Array.prototype.slice.call(arguments);
```
Comment (ivansandrk) [inline common/js/src/logging/logging.js]: What does this line do? Ie. I'm unfamiliar with the Array.proto... syntax.

**Comment by ivansandrk** [comment]
lgtm (+ a question from me)

File: common/js/src/logging/logging.js (hunk 1 above)
Comment (emaxx-google) [inline common/js/src/logging/logging.js]: This is a trick to transform arguments variable (which is for some reason not an array) into an array. I believe it's a pretty common way of doing this, e.g. see the sample code in the Array.slice docs in MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#Array-like_objects

**Pull request merged**